src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/stores/ToolbarStore.js

```diff
@@ -1,17 +1,56 @@
 // @flow
+import type {BackButtonType, DefaultButtonType, DropdownButtonType, ToolbarConfig} from '../types';
 import {action, observable} from 'mobx';
-import type {Item} from '../types';
+
+const defaultConfig = {
+    icons: [],
+    locale: null,
+    buttons: [],
+    backButton: null,
+};
 
 class ToolbarStore {
-    @observable items: Array<Item> = [];
+    @observable config: ToolbarConfig = defaultConfig;
+
+    @action setConfig(config: ToolbarConfig) {
+        this.clearConfig();
+        this.config = {...defaultConfig, ...config};
+    }
+
+    @action clearConfig() {
+        this.config = defaultConfig;
+    }
+
+    hasBackButtonConfig(): boolean {
+        return !!this.config.backButton;
+    }
+
+    getBackButtonConfig(): ?BackButtonType {
+        return this.config.backButton || null;
+    }
+
+    hasButtonsConfig(): boolean {
+        return !!this.config.buttons && !!this.config.buttons.length;
```

Comment:
Is the `length` check necessary? Didn't we have a discussion about that last week?